### ISSUE TITLE
Starting from version 0.4.6, it stopped running with the Parallel plugin

### ISSUE THREAD

**Issue by svk77** [Rosalie241/RMG#166]
Ubuntu 22.04.3 LTS
Here's an error: 
CoreStartEmulation m64p::Core.DoCommand(M64CMD_EXECUTE) Failed: PLUGIN_FAIL: A plugin function returned a fatal error

Perhaps this is due to this innovation in this version?
Add native vulkan support (video plugins can now use vulkan natively without having to use OpenGL for presentation, see https://github.com/mupen64plus/mupen64plus-core/pull/1022)

**Comment by Rosalie241** [comment]
Please provide me with the log from View -> Log when it fails.

**Comment by svk77** [comment]
```
[CORE]  [INFO]    Goodname: Killer Instinct Gold (U) (V1.2) [!]
[CORE]  [INFO]    Name: Killer Instinct Gold
[CORE]  [INFO]    MD5: DD0A82FCC10397AFB37F12BB7F94E67A
[CORE]  [INFO]    CRC: F908CA4C 36464327
[CORE]  [INFO]    Imagetype: .z64 (native)
[CORE]  [INFO]    Rom size: 12582912 bytes (or 12 Mb or 96 Megabits)
[CORE]  [INFO]    Version: 1444
[CORE]  [INFO]    Manufacturer: Nintendo
[CORE]  [INFO]    Country: USA
[CORE]  [INFO]    input plugin did not specify a render callback; there will be no on screen display by the input plugin.
[CORE]  [INFO]    Using video capture backend: dummy
[CORE]  [INFO]    Game controller 0 (Standard controller) has a Memory pak plugged in
[CORE]  [INFO]    Game controller 1 (Standard controller) has a Memory pak plugged in
[CORE]  [INFO]    Game controller 2 (Standard controller) has a Memory pak plugged in
[CORE]  [INFO]    Game controller 3 (Standard controller) has a Memory pak plugged in
[CORE]  [INFO]    Using CIC type X102
[GFX]   [INFO]    Layer count: 3
[GFX]   [INFO]    Found layer: VK_LAYER_MESA_device_select.
[GFX]   [INFO]    Found layer: VK_LAYER_MESA_overlay.
[GFX]   [INFO]    Found layer: VK_LAYER_INTEL_nullhw.
[GFX]   [INFO]    Enabling instance extension: VK_EXT_debug_utils.
[GFX]   [INFO]    Found Vulkan GPU: AMD Radeon RX Vega (RADV VEGA10)
[GFX]   [INFO]        API: 1.3.255
[GFX]   [INFO]        Driver: 23.2.0
[GFX]   [INFO]    Found Vulkan GPU: llvmpipe (LLVM 16.0.6, 256 bits)
[GFX]   [INFO]        API: 1.3.255
[GFX]   [INFO]        Driver: 0.0.1
[GFX]   [INFO]    Using Vulkan GPU: AMD Radeon RX Vega (RADV VEGA10)
[GFX]   [INFO]    Enabling device extension: VK_KHR_swapchain.
[GFX]   [INFO]    Enabling device extension: VK_KHR_sampler_mirror_clamp_to_edge.
[GFX]   [INFO]    Enabling device extension: VK_KHR_external_semaphore_fd.
[GFX]   [INFO]    Enabling device extension: VK_KHR_external_memory_fd.
[GFX]   [INFO]    Enabling device extension: VK_KHR_draw_indirect_count.
[GFX]   [INFO]    Enabling device extension: VK_EXT_calibrated_timestamps.
[GFX]   [INFO]    Enabling device extension: VK_EXT_conservative_rasterization.
[GFX]   [INFO]    Enabling device extension: VK_KHR_image_format_list.
[GFX]   [INFO]    Enabling device extension: VK_KHR_shader_float_controls.
[GFX]   [INFO]    Enabling device extension: VK_KHR_create_renderpass2.
[GFX]   [INFO]    Enabling device extension: VK_KHR_8bit_storage.
[GFX]   [INFO]    Enabling device extension: VK_KHR_16bit_storage.
[GFX]   [INFO]    Enabling device extension: VK_KHR_shader_float16_int8.
[GFX]   [INFO]    Enabling device extension: VK_EXT_subgroup_size_control.
[GFX]   [INFO]    Enabling device extension: VK_NV_compute_shader_derivatives.
[GFX]   [INFO]    Enabling device extension: VK_EXT_host_query_reset.
[GFX]   [INFO]    Enabling device extension: VK_EXT_shader_demote_to_helper_invocation.
[GFX]   [INFO]    Enabling device extension: VK_EXT_scalar_block_layout.
[GFX]   [INFO]    Enabling device extension: VK_KHR_uniform_buffer_standard_layout.
[GFX]   [INFO]    Enabling device extension: VK_KHR_timeline_semaphore.
[GFX]   [INFO]    Enabling device extension: VK_EXT_descriptor_indexing.
[GFX]   [INFO]    Enabling device extension: VK_EXT_memory_priority.
[GFX]   [INFO]    Enabling device extension: VK_EXT_memory_budget.
[GFX]   [INFO]    Enabling device extension: VK_KHR_synchronization2.
[GFX]   [INFO]    Enabling device extension: VK_EXT_pipeline_creation_cache_control.
[GFX]   [INFO]    Enabling device extension: VK_KHR_format_feature_flags2.
[GFX]   [INFO]    Enabling device extension: VK_KHR_buffer_device_address.
[GFX]   [INFO]    Enabling device extension: VK_KHR_spirv_1_4.
[GFX]   [INFO]    Enabling device extension: VK_KHR_shader_subgroup_extended_types.
[GFX]   [INFO]    Enabling device extension: VK_KHR_present_id.
[GFX]   [INFO]    Enabling device extension: VK_KHR_present_wait.
[GFX]   [INFO]    Enabling device extension: VK_EXT_external_memory_host.
[GFX]   [ERROR]   Failed to create VkSurfaceKHR.
[GFX]   [ERROR]   Could not start GFX plugin
[CORE]  [STATUS]  Rom closed.
```

**Comment by Rosalie241** [comment]
Can you try running RMG from the commandline with the `RMG_SHOW_DEBUG_QT_MESSAGES=1` variable set and then give me the commandline output when it fails?

**Comment by svk77** [comment]
I don't understand how to do it with appimage package

**Comment by metchebe** [comment]
Hi, first of all thank you for this project.

I have the same error on Kubuntu 22.04.3 with RMG v0.5.0 built from source. GPU is CoffeeLake-S GT2 [UHD Graphics 630]. As far as I can tell, Vulkan works ok on this machine (tested with other apps).

Everything else except the parallel video plugin works OK.

Here are the QT messages:

```
metchebe@optiplex:~/Applications/RMG$ RMG_SHOW_DEBUG_QT_MESSAGES=1 ./RMG 
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_EventFilter_KeyPressed(QKeyEvent*)
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_EventFilter_KeyReleased(QKeyEvent*)
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_EventFilter_FileDropped(QDropEvent*)
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_QGuiApplication_applicationStateChanged(Qt::ApplicationState)
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_System_OpenRom()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_System_OpenCombo()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_System_Shutdown()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_System_SoftReset()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_System_HardReset()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_System_Pause()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_System_Screenshot()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_System_LimitFPS()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_System_SpeedFactor(int)
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_System_SaveState()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_System_SaveAs()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_System_LoadState()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_System_Load()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_System_CurrentSaveState(int)
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_System_Cheats()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_System_GSButton()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_System_Exit()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_Settings_Graphics()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_Settings_Audio()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_Settings_Rsp()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_Settings_Input()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_Settings_Settings()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_View_Toolbar(bool)
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_View_StatusBar(bool)
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_View_GameList(bool)
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_View_GameGrid(bool)
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_View_UniformSize(bool)
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_View_Fullscreen()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_View_RefreshRoms()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_View_ClearRomCache()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_View_Log()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_Help_Github()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_Help_About()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_Help_Update()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_Audio_IncreaseVolume()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_Audio_DecreaseVolume()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Action_Audio_ToggleVolumeMute()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Emulation_Started()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Emulation_Finished(bool)
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_RomBrowser_PlayGame(QString)
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_RomBrowser_PlayGameWith(CoreRomType,QString)
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_RomBrowser_ChangeRomDirectory()
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_RomBrowser_RomInformation(QString)
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_RomBrowser_EditGameSettings(QString)
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_RomBrowser_EditGameInputSettings(QString)
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_RomBrowser_Cheats(QString)
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_VidExt_Init(VidExtRenderMode)
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_VidExt_SetupOGL(QSurfaceFormat,QThread*)
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_VidExt_SetWindowedMode(int,int,int,int)
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_VidExt_SetFullscreenMode(int,int,int,int)
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_VidExt_ResizeWindow(int,int)
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_VidExt_ToggleFS(bool)
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Core_DebugCallback(CoreDebugMessageType,QString,QString)
[QT WARNING] QMetaObject::connectSlotsByName: No matching signal for on_Core_StateCallback(CoreStateCallbackType,int)
[QT WARNING] Failed to load vulkan: Cannot load library vulkan: (vulkan: cannot open shared object file: No such file or directory)
[QT WARNING] initInstance: No Vulkan library available
[QT WARNING] Failed to create platform Vulkan instance
[QT WARNING] Attempted to create Vulkan surface without an instance; was QWindow::setVulkanInstance() called?
Segmentation fault (core dumped)
```

**Comment by Rosalie241** [comment]
Can you try running RMG with `QT_VULKAN_LIB=libvulkan.so.1`? i.e `QT_VULKAN_LIB=libvulkan.so.1 ./RMG`

**Comment by metchebe** [comment]
`QT_VULKAN_LIB=libvulkan.so.1 ./RMG` works!

Thank you.

**Comment by svk77** [comment]
QT_VULKAN_LIB=libvulkan.so.1
This works with appimage too
Thank you

**Comment by Rosalie241** [comment]
I've pushed some commits to master which fix this (https://github.com/Rosalie241/RMG/commit/38b946926a735a49f4a812fd7413c28b519d0e71 and https://github.com/Rosalie241/RMG/commit/422f7d08015d8964d39b878ad2a4b6807bb64f9a) so you wont need to set `QT_VULKAN_LIB` manually anymore, please try the following CI build: https://github.com/Rosalie241/RMG/actions/runs/5986345379

**Comment by Rosalie241** [comment]
The fix has been included in v0.5.1, thank you for the report!

**Comment by metchebe** [comment]
Yes, fixed indeed, thank you very much!
